Reslice mapper representation

sankhesh opened the pull request:
Exposes the vtk-js vtkImageResliceMapper as a representation in react-vtk-js

sankhesh committed:
vtk-js: Update version number

sankhesh committed:
feat(Representation): Added a new ResliceRepresentation

sankhesh committed:
test(Rendering): Added an example for ResliceRepresentation

sankhesh committed:
fix(ResliceRepresentation): Ensure that the slice function is set

sankhesh committed:
test(ResliceRepresentation): Example for switchable slice function

sankhesh commented:
Depends on https://github.com/Kitware/vtk-js/pull/2839

sankhesh commented:
@floryst @jadh4v

floryst commented:
Overall LGTM. Still haven't tested it. Will port over to beta when this is merged.

floryst commented:
I see 2839 has been merged. Once this PR has been updated, I will test it out.

sankhesh commented:
Do I need to modify the minimum vtk-js version or will the latest be automatically pulled? The new release puts vtk-js at [v28.2.1](https://github.com/Kitware/vtk-js/releases/tag/v28.2.1)

floryst commented:
The package-lock.json will need to be updated with an `npm install`. It's currently on 28.2.0.

Also, I just realized this is a breaking change, since we are requiring a new major version of vtk.js as a dependency. In that case, I'll rewrite this with your input onto the beta branch. Once that's done, we can close this instead of merging.

sankhesh committed:
refactor(ResliceRepresentation): Migrate to functional component with TypeScript

Convert from class-based JS to functional TSX component with proper type definitions. Update build scripts and exports accordingly.

sankhesh committed:
Add missing ShareDataSet for backwards compatibility

ShareDataSet was being used in ResliceRendering but not exported.

sankhesh committed:
Fix RegisterDataSet to re-evaluate connection on dataAvailable

sankhesh committed:
Fix ResliceRepresentation mapper recreated on every render

sankhesh committed:
Fix ResliceRendering example for updated API

sankhesh committed:
Add orientation combo-box to ResliceRendering example

sankhesh committed:
Fix camera reset on every render by memoizing camera props

sankhesh committed:
Revert "Fix camera reset on every render by memoizing camera props"

This reverts commit 9fe333bb7a6ed894ad37821d07f093ffd6ed8641.

sankhesh committed:
Add missing dev dependencies for ESLint and Prettier plugins

sankhesh committed:
Revert package changes

jadh4v commented on src/core/ResliceRepresentation.tsx:
```diff
@@ -0,0 +1,148 @@
+  return (
+    <SliceRepresentation
+      {...sliceProps}
+      mapperInstance={mapperInstance as any}
```
Why is `any` needed here?
Can it be fixed by using `vtkAbstractImageMapper` in `SliceRepresentation` at line 55?
change `mapperInstance?: AbstractImageMapper;` to `mapperInstance?: vtkAbstractImageMapper;`

jadh4v commented on src/core/ResliceRepresentation.tsx:
```diff
@@ -0,0 +1,148 @@
+  /**
+   * Slice plane
+   */
+  slicePlane?: any;
```
Is this a vtkPlane?
```suggestion
  slicePlane?: vtkPlane;
```

jadh4v commented on src/core/ResliceRepresentation.tsx:
```diff
@@ -0,0 +1,148 @@
+  /**
+   * Optional slice polydata
+   */
+  slicePolyData?: any;
```
```suggestion
  slicePolyData?: vtkPolyData;
```

jadh4v commented on src/core/ResliceRepresentation.tsx:
```diff
@@ -0,0 +1,148 @@
+
+  function trackModified(changed: boolean) {
+    if (changed) {
+      // Trigger render
```
Do we leave this implementation empty for now?
Here's what I got from Claude AI:

_It does nothing. But in the parent SliceRepresentation, trackModified comes from useBooleanAccumulator() and is what actually drives renderer.requestRender(). So every trackModified(true) call in the five reslice effects (slicePlane, slicePolyData, slabType, slabThickness, slabTrapezoidIntegration) is dead code with a comment promising a render that never happens.
The practical consequence: the effects do call mapperInstance.setSlabThickness(...) etc., so the mapper object is updated — but no render is requested. And because the reslice-specific props are destructured out and not forwarded to <SliceRepresentation> (they go into the discarded reslice locals, only ...sliceProps is passed down), a change to, say, slabThickness alone won't cause the child to re-render either. So changing those props at runtime produces no visible update until some unrelated interaction forces a render. Worth testing directly — it lines up with floryst's note that he hasn't actually tested it yet.
The fix is to thread the real trackModified/render-request mechanism through instead of shadowing it with a stub._
